addons/payment_xendit/static/src/js/payment_form.js

```diff
@@ -0,0 +1,212 @@
+        if (providerCode !== 'xendit' || paymentMethodCode != 'card') {
+            this._super(...arguments);
+            return;
+        }
+
+        // Check if the inline form values were already extracted.
+        this.xenditData ??= {}; // Store the form data of each instantiated payment method.
+        if (flow === 'token') {
+            return; // Don't show the form for tokens.
+        } else if (this.xenditData[paymentOptionId]) {
+            this._setPaymentFlow('direct'); // Overwrite the flow even if no re-instantiation.
+            return; // Don't re-extract the data if already done for this payment method.
+        }
+
+        // Overwrite the flow of the selected payment method.
+        this._setPaymentFlow('direct');
+
+        // Extract and deserialize the inline form values.
+        const radio = document.querySelector('input[name="o_payment_radio"]:checked');
+        const inlineForm = this._getInlineForm(radio);
+        const xenditForm = inlineForm.querySelector('[name="o_xendit_form"]');
+        this.xenditData[paymentOptionId] = JSON.parse(
+            xenditForm.dataset['xenditInlineFormValues']
+        );
+        this.xenditData[paymentOptionId].form = xenditForm;
+
+        // Load the SDK.
+        await loadJS("https://js.xendit.co/v1/xendit.min.js");
+
+        Xendit.setPublishableKey(this.xenditData[paymentOptionId]['public_key'])
+    },
+
+    // #=== PAYMENT FLOW ===#
+
+    /**
+     * Trigger the payment processing by submitting the data.
+     *
+     * @override method from payment.payment_form
+     * @private
+     * @param {string} providerCode - The code of the selected payment option's provider.
+     * @param {number} paymentOptionId - The id of the selected payment option.
+     * @param {string} paymentMethodCode - The code of the selected payment method, if any.
+     * @param {string} flow - The payment flow of the selected payment option.
+     * @return {void}
+     */
+    async _initiatePaymentFlow(providerCode, paymentOptionId, paymentMethodCode, flow) {
+        if (providerCode !== 'xendit' || flow === 'token' || paymentMethodCode != 'card') {
+            this._super(...arguments); // Tokens are handled by the generic flow
+            return;
+        }
+        const formInputs = this._xenditGetInlineFormInputs(paymentOptionId, paymentMethodCode)
+        const details = this._xenditGetPaymentDetails(paymentOptionId, paymentMethodCode)
+
+        Object.keys(formInputs).forEach(el => formInputs[el].setCustomValidity(""))
+        if(!Xendit.card.validateCardNumber(details.card_number)){
+            formInputs['card'].setCustomValidity("Invalid Input")
+        }
+        if(!Xendit.card.validateExpiry(details.card_exp_month, details.card_exp_year)){
+            formInputs['month'].setCustomValidity("Invalid Input")
+            formInputs['year'].setCustomValidity("Invalid Input")
```

Comment:
@chevalierv for this one, I think I'll do `_t("Invalid Date")` for both form inputs because suppose date is (10/2022), the invalid one is year and it would be pretty misleading to say month is invalid. Let me know if you have a better sugggestion

Reply:
LGTM